src/services/cohortInfos.ts

```diff
@@ -147,37 +149,26 @@ const fetchCohort = async (cohortId: string | undefined): Promise<CohortData | u
 
   if (CONTEXT === 'arkhn') {
     const cohortResult: CohortData = {}
-    const cohort = getApiResponseResources(
-      await api.get<FHIR_API_Response<IGroup>>(`/Group?_id=${cohortId}${API_RESOURCE_TAG}`)
-    )?.[0]
-
-    if (cohort) {
-      cohortResult.cohort = cohort
-
-      //Fetch cohort related patients & encounters
-      const patientIdentifiers = cohort.member
-        ?.map((m) => m.entity.identifier?.value)
-        .filter((id): id is string => undefined !== id)
-      const patientRefs = cohort.member
-        ?.map((m) => m.entity.reference)
-        .filter((ref): ref is string => undefined !== ref)
-
-      if (!patientIdentifiers || !patientRefs) {
-        return cohortResult
-      }
+    const response = getApiResponseResources(
+      await api.get<FHIR_API_Response<IGroup | IPatient>>(`/Group?_id=${cohortId}&_include=Group:member`)
+    )
+    if (response) {
+      cohortResult.cohort = head(response.filter((resource) => resource.resourceType === 'Group') as IGroup[])
 
-      const [patientsResp, encountersResp] = await Promise.all([
-        api.get<FHIR_API_Response<IPatient>>(`/Patient?identifier=${patientIdentifiers.join(',')}${API_RESOURCE_TAG}`),
-        api.get<FHIR_API_Response<IEncounter>>(`/Encounter?subject=${patientRefs.join(',')}${API_RESOURCE_TAG}`)
-      ])
-      const patients = getApiResponseResources(patientsResp)
+      const patients = response.filter((resource) => resource.resourceType === 'Patient') as IPatient[]
+
+      if (!patients) return cohortResult
+
+      const encountersResp = await api.get<FHIR_API_Response<IEncounter>>(
+        `/Encounter?subject=${patients.map((patient) => patient.id).join(',')}`
+      )
```

Comment:
I think we can also avoid this query: https://www.hl7.org/fhir/search.html#revinclude (look at the :iterate modifier)
```
GET [base]/MedicationDispense?_include=MedicationDispense:prescription
    &_include:iterate=MedicationRequest:performer&criteria...
```
don't know if this is supported by hapi though